core/server/common/src/main/java/alluxio/master/journal/raft/BufferedJournalApplier.java

```diff
@@ -309,5 +310,20 @@ protected void runCatchup() {
         }
       }
     }
+
+    @Override
+    public void onError(Throwable t) {
```

Comment:
The implementations of `onError()` for `RaftJournalCatchupThread` and `UfsJournalCatchupThread` are identical. Why not change the default implementation in `AutopsyThread` to match the one in those two classes (while removing the `ProcessUtils.fatalError` call), and selectively override `onError()` only in `UfsJournalCheckpointThread`? 
With these changes you could get rid of the `abstract` keyword for `AutopsyThread`, `onError()`, and the `UfsJournalCheckpointThread#onError()` could simply be 
```
@Override
public void onError(Throwable t) {
  mCatchupState = CatchupState.DONE;
  super.onError(t);
}
```

Reply:
I chose to be verbose and write down all the code in each implementation so that we can scrutinize and discuss whether each one is necessary. Now that I've got my discussion and the last merging step will be trivial.